### DISCUSSION TITLE
Incorporating recent nuget features into the NuGet Package template

### DISCUSSION THREAD

**Discussion by kimsagro1** [Dotnet-Boxed/Templates#1702] [Ideas]
Recently Claire Novotny, the PM on the NuGet team at Microsoft posted a video talking about some [recent nuget features](https://www.youtube.com/watch?v=C_2BStepVKw) such as

- Package validation
- The [DotNet.ReproducibleBuilds](https://github.com/dotnet/reproducible-builds) package

I thought these could be potentially introduced to the NuGet Package template

**Comment by RehanSaeed** [comment]
I saw that talk and have followed up with questions about [DotNet.ReproducibleBuilds](https://github.com/dotnet/reproducible-builds) as its not well documented.

- https://github.com/dotnet/sdk/issues/2679#issuecomment-1197945529

I'm not sure it uses the best defaults but agree that we should switch over to it.

**Comment by RehanSaeed** [comment]
Package validation is a cool feature. The downside right now is that you have to pass it the previous version number for it to compare to. Ideally, it'd do that for you. I have a GitHub issue for that somewhere.

**Comment by RehanSaeed** [comment]
I found the relevant issue. Until this is implemented, I package validation requires a lot of work to build into the cake file.

- https://github.com/dotnet/sdk/issues/25268

**Comment by RehanSaeed** [comment]
FYI: [DotNet.ReproducibleBuilds](https://github.com/dotnet/reproducible-builds) was added to the template.
